eslint.config.mjs

```diff
@@ -0,0 +1,127 @@
+import eslint from '@eslint/js'
+import vitest from '@vitest/eslint-plugin'
+import importPlugin from 'eslint-plugin-import'
+import jestDom from 'eslint-plugin-jest-dom'
+import react from 'eslint-plugin-react'
+import reactCompiler from 'eslint-plugin-react-compiler'
+import reactHooks from 'eslint-plugin-react-hooks'
+import testingLibrary from 'eslint-plugin-testing-library'
+import globals from 'globals'
+import tseslint from 'typescript-eslint'
+
+export default tseslint.config(
+  {
+    ignores: ['**/dist/'],
+  },
+  eslint.configs.recommended,
+  tseslint.configs.recommended,
+  react.configs.flat.recommended,
+  react.configs.flat['jsx-runtime'],
+  {
+    languageOptions: {
+      globals: {
+        ...globals.browser,
+        ...globals.node,
+        ...globals.es2015,
+      },
+      parser: tseslint.parser,
+      parserOptions: {
+        ecmaVersion: 2018,
+        sourceType: 'module',
+        ecmaFeatures: {
+          jsx: true,
+        },
+      },
```

Comment:
As we use `tseslint.config`, can it be just default?

Reply:
* https://github.com/import-js/eslint-plugin-import/blob/main/config/recommended.js#L24-L27
* https://github.com/typescript-eslint/typescript-eslint/blob/main/packages/typescript-eslint/src/configs/base.ts#L13-L16
* https://github.com/jsx-eslint/eslint-plugin-react/blob/master/index.js#L41-L44
* https://github.com/jsx-eslint/eslint-plugin-react/blob/master/index.js#L82-L87

@dai-shi The properties specified in the link are all merged, so they are the same as the ones currently set in `eslint.config.mjs`. Since it is a duplicate configuration, I will remove it.

Reply:
```suggestion
      parserOptions: {
        ecmaVersion: 2018,
      },
```

@dai-shi I think we can change this like this

Reply:
looks good. what's the default `ecmaVersion` if we don't specify?

Reply:
@dai-shi It seems that `ecmaVersion` does not have a default value as of now

Reply:
```suggestion
```

https://eslint.org/docs/latest/use/configure/language-options#specifying-javascript-options

@dai-shi In this document `ecmaVersion` default value is `latest`

Reply:
yeah, we are transpiling anyway. let's use the default `latest` for now until we really find any issues.

Reply:
https://github.com/import-js/eslint-plugin-import/blob/main/config/recommended.js#L24-L27

@dai-shi However, `importPlugin.flatConfigs.recommended` sets `ecmaVersion: 2018`, so if we follow this, it seems we can set `ecmaVersion` accordingly.

Reply:
> yeah, we are transpiling anyway. let's use the default `latest` for now until we really find any issues.

@dai-shi We can set it to `ecmaVersion: 2018`, but do you want to use `latest` for now until any issues arise?

Reply:
> However, `importPlugin.flatConfigs.recommended` sets `ecmaVersion: 2018`,

If that's the case, I change my mind from https://github.com/pmndrs/zustand/pull/2912#discussion_r1897397153, and suggest to use `importPlugin.flatConfigs.recommended` and make some rules `off`.

Reply:
> > However, `importPlugin.flatConfigs.recommended` sets `ecmaVersion: 2018`,
> 
> If that's the case, I change my mind from [#2912 (comment)](https://github.com/pmndrs/zustand/pull/2912#discussion_r1897397153), and suggest to use `importPlugin.flatConfigs.recommended` and make some rules `off`.

@dai-shi Ok, i'll reflect your suggest